state/protocol/badger/state.go

```diff
@@ -164,6 +170,21 @@ func (state *State) bootstrapSealingSegment(segment *flow.SealingSegment, head *
 			if err != nil {
 				return fmt.Errorf("could not index execution result: %w", err)
 			}
+
+			// SealingSegment might not have all the blocks referenced by results.
+			// If the block is missing we skip it while indexing, and that's OK because
+			// we are bootstrapping and service event index before the segment shouldn't matter
+			block, has := blocksById[result.BlockID]
+			if !has {
+				continue
+			}
+
+			for _, serviceEvent := range result.ServiceEvents {
+				err := transaction.WithTx(operation.IndexExecutionResultByServiceEventTypeAndHeight(result.ID(), serviceEvent.Type, block.Header.Height))(tx)
```

Comment:
This is indexing the old way - now it isn't consistent with how service events are indexed when finalized.

Reply:
Adding more context:

Given a segment of blocks like:
```
A<-B(Result_A)<-C(Seal_A)<-D(QC_C)
```

In the bootstrapping here, we are indexing a service event emitted in block A, by A.height.

However, when we process blocks while the process is running ([here](https://github.com/onflow/flow-go/pull/3605/files#diff-c79d3fc5d32f153917b7cd923ab4ba64ef22f12268ddd13f353a06391f68a96dR574-R586)) we are indexing by D.height , like we discussed. 

So we are indexing inconsistently between bootstrapping and ongoing block processing. 

To index the same way in both places, we would need to look through the seals contained in the block paylods of the sealing segment used to bootstrap. In particular, if the sealing segment contains an execution result where a service event was emitted, but the result hasn't been sealed yet, we shouldn't index it during bootstrapping.

For example, if `Result_X` has a service event, and we're processing the sealing segment `W<-X<-Y<-Z` during bootstrapping, we should not index that service event yet, since it hasn't been sealed.
```
W<-X(Result_W)<-Y(Result_X)<-Z(Seal_W)
```